fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/RankIndexMaintainer.java

```diff
@@ -116,7 +116,7 @@ public RankIndexMaintainer(IndexMaintainerState state) {
         final List<CompletableFuture<Void>> futures = new ArrayList<>();
         for (IndexEntry indexEntry : indexEntries) {
             // First maintain an ordinary B-tree index by score.
-            updateOneKey(savedRecord, remove, indexEntry);
+            CompletableFuture<Void> updateOrdinaryIndex = updateOneKeyAsync(savedRecord, remove, indexEntry);
```

Comment:
Is there a reason to not do something like:

```java
if (!MoreAsyncUtil.isCompletedNormally(updateOrdinaryIndex)) {
    futures.add(updateOridnaryIndex);
}
```

Then I think it would look for uniqueness violations in parallel with updating the ranked set. (And the check for if it's completed normally is just to avoid waiting on done futures, but it should be optional.)

I didn't leave a comment about it, but I think the same thing applies to the `TimeWindowLeaderboardIndexMaintainer`.